holidays/generate_ics.py

```
@@ -194,9 +224,30 @@ def run(self) -> None:
+                "start_year": start_year,
+                "end_year": end_year,
+                "today": datetime.now(timezone.utc).strftime("%Y%m%d"),
+            }
+            try:
+                output_path = template.format(**values)
+            except KeyError as e:
+                supported = ", ".join(f"{{{k}}}" for k in values)
+                raise SystemExit(
+                    f"Unknown placeholder '{{{e.args[0]}}}' in output template. "
+                    f"Supported placeholders: {supported}"
+                )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)